tests/test_utils.py

```diff
@@ -66,6 +66,32 @@ def test_file(self, tmpdir, mode, warnings_num, recwarn):
             assert super_len(fd) == 4
         assert len(recwarn) == warnings_num
 
+    def test_super_len_with__len__(self):
+        foo = [1,2,3,4]
+        len_foo = super_len(foo)
+        assert len_foo == 4
+
+    def test_super_len_with_no__len__(self):
+        foo = BytesIO(b'1234')
+        assert getattr(foo, '__len__', None) is None
+        setattr(foo, 'len', 4)
```

Comment:
@Lukasa do you know of a class with `len` but no `__len__` this works but is a little hacky.